cpp/src/arrow/compute/exec/exec_plan.cc

```diff
@@ -157,11 +157,50 @@ struct ExecPlanImpl : public ExecPlan {
     return std::move(Impl{nodes_}.sorted);
   }
 
+  std::pair<NodeVector, std::vector<int>> OrderedNodes() const {
```

Comment:
What does this do, is this different from the topological sort above? In any case, please add a comment describing this (what does it return?).